### PR TITLE
(CONT-796) Puppet 8 support

### PR DESCRIPTION
**Pull request by LukasAud**
This PR aims to update the module with compatibility for the upcoming Puppet 8 agent release. The most relevant changes introduced in this updated are:
Updated version requirements in metadata.json
Bumped Rubocop gems
Adjusted Rubocop configuration
Resolved new Rubocop offenses

### CODE REVIEW

File: .fixtures.yml [hunk 1]
```diff
@@ -2,9 +2,9 @@ fixtures:
   repositories:
     facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
     provision: 'https://github.com/puppetlabs/provision.git'
-    puppet_agent:
+    puppet_agent:
```
Comment (jordanbreen28) [inline .fixtures.yml:5]: some whitespace

File: .fixtures.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@ fixtures:
   repositories:
     facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
     provision: 'https://github.com/puppetlabs/provision.git'
-    puppet_agent:
+    puppet_agent: 
       repo: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
-      ref: v4.12.1
+      ref:  v4.13.0
```
Comment (jordanbreen28) [inline .fixtures.yml:7]: same as above

File: lib/puppet/type/registry_key.rb [hunk 3]
```diff
@@ -128,4 +128,6 @@ def eval_generate
     end
     resources
   end
+  # rubocop:enable Metrics/MethodLength
```
Comment (jordanbreen28) [inline lib/puppet/type/registry_key.rb:131]: could probably remove `# rubocop:enable Metrics/MethodLength ` here as its duplicated in L133

File: .fixtures.yml (hunk 2 above)
Comment (LukasAud) [reply .fixtures.yml:7]: will add

File: lib/puppet/provider/registry_value/registry.rb [hunk 4]
```diff
@@ -260,3 +270,4 @@ def path
     @path ||= PuppetX::Puppetlabs::Registry::RegistryValuePath.new(resource.parameter(:path).value)
   end
 end
+# rubocop:enable Metrics/BlockLength
```
[the remark this answers is not in the document]
Comment (LukasAud) [reply lib/puppet/provider/registry_value/registry.rb:273]: This is a blocklength being disabled much further up.

File: lib/puppet/type/registry_key.rb (hunk 3 above)
Comment (LukasAud) [reply lib/puppet/type/registry_key.rb:131]: again, methodlength vs blocklength

File: .fixtures.yml (hunk 1 above)
Comment (LukasAud) [reply .fixtures.yml:5]: ah I see, no worries.. will fix

File: lib/puppet/provider/registry_value/registry.rb (hunk 4 above)
[the remark this answers is not in the document]
Comment (jordanbreen28) [reply lib/puppet/provider/registry_value/registry.rb:273]: the cop is enabled again on L231 and not disabled again by the looks of it
https://github.com/puppetlabs/puppetlabs-registry/blob/46c7589f29b3b932be49b726d190f5bfb2dad66c/lib/puppet/provider/registry_value/registry.rb#L231

[the remark this answers is not in the document]
Comment (jordanbreen28) [reply lib/puppet/provider/registry_value/registry.rb:273]: Nvm... misread on my part!

**Review by david22swan — APPROVED** [review]
LGTM

**Pull request merged**